src/modules/claims/claims.service.ts

```diff
@@ -546,10 +530,23 @@ export class ClaimsService {
       claimTypeVersion,
       claimType,
       acceptedBy,
-      subjectAgreement,
       onChainProof,
       expirationTimestamp,
     } = claim;
+    let { subjectAgreement } = claim;
+
+    if (await this.hasOnChainRole(subject, claimType, +claimTypeVersion)) {
+      return;
```

Comment:
I think it would be good to add a log line here, maybe at the `info` level.

Reply:
I think warning is more appropriate, because it is unexpected